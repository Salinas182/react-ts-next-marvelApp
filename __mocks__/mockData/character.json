{
  "code": 200,
  "status": "Ok",
  "copyright": "© 2024 MARVEL",
  "attributionText": "Data provided by Marvel. © 2024 MARVEL",
  "attributionHTML": "<a href=\"http://marvel.com\">Data provided by Marvel. © 2024 MARVEL</a>",
  "etag": "1e70aff2da6bcf86aa29ff99ce19645a20281317",
  "data": {
    "offset": 0,
    "limit": 20,
    "total": 1,
    "count": 1,
    "results": [
      {
        "id": 1010727,
        "name": "Spider-dok",
        "description": "",
        "modified": "1969-12-31T19:00:00-0500",
        "thumbnail": {
          "path": "http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available",
          "extension": "jpg"
        },
        "resourceURI": "http://gateway.marvel.com/v1/public/characters/1010727",
        "comics": {
          "available": 0,
          "collectionURI": "http://gateway.marvel.com/v1/public/characters/1010727/comics",
          "items": [],
          "returned": 0
        },
        "series": {
          "available": 0,
          "collectionURI": "http://gateway.marvel.com/v1/public/characters/1010727/series",
          "items": [],
          "returned": 0
        },
        "stories": {
          "available": 0,
          "collectionURI": "http://gateway.marvel.com/v1/public/characters/1010727/stories",
          "items": [],
          "returned": 0
        },
        "events": {
          "available": 0,
          "collectionURI": "http://gateway.marvel.com/v1/public/characters/1010727/events",
          "items": [],
          "returned": 0
        },
        "urls": [
          {
            "type": "detail",
            "url": "http://marvel.com/characters/2170/spider-dok?utm_campaign=apiRef&utm_source=3137067949a68b4ca8c4db83b4f42475"
          },
          {
            "type": "comiclink",
            "url": "http://marvel.com/comics/characters/1010727/spider-dok?utm_campaign=apiRef&utm_source=3137067949a68b4ca8c4db83b4f42475"
          }
        ]
      }
    ]
  }
}
