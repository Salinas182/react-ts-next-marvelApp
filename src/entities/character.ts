export interface Character {
  id: number;
  name: string;
  description: string;
  modified: string;
  thumbnail: Thumbnail;
  resourceURI: string;
  comics: Comics;
  series: Comics;
  stories: Stories;
  events: Comics;
  urls: URL[];
}

export interface Collections {
  available: number;
  collectionURI: string;
  returned: number;
}

export interface Comics extends Collections {
  items: CollectionsItem[];
}

export interface Series extends Collections {
  items: CollectionsItem[];
}

export interface CollectionsItem {
  resourceURI: string;
  name: string;
}

export interface Stories {
  available: number;
  collectionURI: string;
  items: StoriesItem[];
  returned: number;
}

export interface StoriesItem extends CollectionsItem {
  type: Type;
}

export enum Type {
  Cover = 'cover',
  InteriorStory = 'interiorStory',
}

export interface Thumbnail {
  path: string;
  extension: string;
}

export interface URL {
  type: string;
  url: string;
}
