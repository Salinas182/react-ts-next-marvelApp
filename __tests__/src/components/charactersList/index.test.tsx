import React from 'react';
import { render, screen } from '@testing-library/react';
import CharactersList from '@/components/charactersList';
import charactersResponse from '../../../../__mocks__/mockData/characters.json';
import { Character } from '@/entities/character';

jest.mock('@/hooks/useFavorites', () => () => ({
  favorites: [] as Character[],
  updateFavorites: jest.fn(),
}));
jest.mock('@/components/characterCard/index.tsx', () => ({
  __esModule: true,
  default: jest.fn(() => <div>Character Card</div>),
}));

const mockCharacters: Character[] = charactersResponse.data.results;

describe('CharactersList', () => {
  it('should render CharacterCard components for each character', () => {
    render(<CharactersList characters={mockCharacters} />);

    expect(screen.getAllByText('Character Card')).toHaveLength(
      mockCharacters.length
    );
  });

  it('should display a message when no characters are available', () => {
    render(<CharactersList characters={[]} />);

    expect(screen.getByText('No characters available')).toBeInTheDocument();
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(
      <CharactersList characters={mockCharacters} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
