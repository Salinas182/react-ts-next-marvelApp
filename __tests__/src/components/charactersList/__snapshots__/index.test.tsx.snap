// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharactersList matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="gridContainer"
  >
    <div
      class="grid"
    >
      <div>
        Character Card
      </div>
      <div>
        Character Card
      </div>
      <div>
        Character Card
      </div>
      <div>
        Character Card
      </div>
      <div>
        Character Card
      </div>
    </div>
  </div>
</DocumentFragment>
`;
